libs/rtemodel/include/RteTarget.h

```diff
@@ -46,15 +46,17 @@ struct RteFileInfo
   /**
    * @brief compare file version of the given target and the instance
    * @param targetName target name
+   * @param rteFolder the "RTE" folder path used for placing files
    * @return 0 if both file versions are same, > 0 if file version of the given target is newer, otherwise < 0
   */
-  int HasNewVersion(const std::string& targetName) const;
+  int HasNewVersion(const std::string& targetName, const std::string& rteFolder) const;
```

Comment:
target knows its project, no need to pass en extra argument

Reply:
I was a bit worried here since the GetProject() may return nullptr sometimes. But looking through the current code it seems an RteTarget is always created with a parent that is at least an RteProject, so GetProject() should always return something in the current code.

If we start relying on this behaviour, perhaps it would make sense to change https://github.com/Open-CMSIS-Pack/devtools/blob/main/libs/rtemodel/include/RteTarget.h#L77 from `RteItem* parent` to `RteProject* parent` to avoid creating a "bad" RteTarget object in the future that would not have any associated project? What do you think?

Reply:
We might change the signature to pass RteProject* has a parent. However it will be anyway stored as RteItem* in the base class.
 GetProject() may return nullptr in packchk executable:
`CheckConditions.cpp(23):m_target` = new RteTarget(nullptr, m_filteredModel, "CondTest", EMPTY_MAP);`

Anyway I would not change that in this PR. 
We already have some ideas on refactoring and improvement in the RTE. 
We can consider your proposal as well.